lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -613,6 +616,44 @@ void FunctionAnalysisResults::importModel() {
   }
 }
 
+void FunctionAnalysisResults::importDynamicSymbols() {
+  llvm::Function &Root = *M.getFunction("root");
+
+  for (BasicBlock &BB : Root) {
+    if (getType(&BB) != BlockType::JumpTargetBlock)
+      continue;
+
+    auto *SymbolNameString = hasDynamicSymbol(&BB);
+    if (not SymbolNameString)
+      continue;
+
+    auto [Entry, _] = getPC(&*BB.begin());
+
+    // Already imported?
+    if (FunctionsBucket.find(Entry) != FunctionsBucket.end())
+      continue;
+
+    llvm::StringRef SymbolName = extractFromConstantStringPtr(SymbolNameString);
+    revng_assert(SymbolName.size() != 0);
+
+    // TODO: change this to assert, dynamic function must have been imported
+    auto It = Binary->ImportedDynamicFunctions.find(SymbolName.str());
+    if (It == Binary->ImportedDynamicFunctions.end())
+      continue;
+
+    if (not It->Prototype.isValid())
+      It->Prototype = Binary->DefaultPrototype;
+
+    model::FunctionType::Values FunctionType;
+    if (It->Attributes.count(model::FunctionAttribute::NoReturn) != 0)
+      FunctionType = model::FunctionType::NoReturn;
+    else
+      FunctionType = model::FunctionType::Regular;
```

Comment:
We really need to drop `model::FunctionType` entirely.
Please put it up next in the task list.